File: modules/compute/vpc.go
package compute

import (
	"pul8s/modules/shared"

	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/ec2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func NewVPC(ctx *pulumi.Context) (*ec2.Vpc, error) {
	azCount := 2

	vpc, err := ec2.NewVpc(ctx, "pul8s", &ec2.VpcArgs{
		NumberOfAvailabilityZones: &azCount,
		Tags:                      shared.NewCommonTags("vpc"),
	})
	if err != nil {
		return nil, err
	}

	return vpc, nil
}

File: modules/compute/vpc.ts
import * as awsx from '@pulumi/awsx';
import { NewResourceName } from '../shared/resource/name';
import { NewResourceTags } from '../shared/resource/tags';

export function NewVpc(azCount: number): awsx.ec2.Vpc {
    return new awsx.ec2.Vpc(NewResourceName("vpc"), {
        numberOfAvailabilityZones: azCount,
        tags: NewResourceTags("vpc"),
    })
}